tests/chainer_tests/functions_tests/evaluation_tests/test_accuracy.py

```diff
@@ -12,6 +13,34 @@
 from chainer.utils import type_check
 
 
+def _testing_pairwise(*parameters):
```

Comment:
I found this function difficult to understand.
Especially it's difficult to track the data structures of each variables (e.g. `parameters`, `product_dicts`, etc.).
Could you annotate the code with approppriate comments/assertions?

Reply:
Sure.  I'll add comments or assertions.